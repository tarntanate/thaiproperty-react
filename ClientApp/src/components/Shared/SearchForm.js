import React, { Component } from 'react';
import { connect } from 'react-redux';
// Total antd module size is 324K gzipped, so we must selectively import
import Card from 'antd/lib/card/';
import Form from 'antd/lib/form/';
import Button from 'antd/lib/button/';
import Input from 'antd/lib/input/';
import Icon from 'antd/lib/icon/';

const FormItem = Form.Item;

class SearchForm extends Component {
  state = {
    search: '',
  };

  handleFormSubmit = e => {
    e.preventDefault();
    this.props.handleFormSubmit();
  };

  render() {
    return (
      <Card bordered={false}>
        <Form layout="inline" onSubmit={this.handleFormSubmit}>
          <FormItem wrapperCol={{ span: 24 }}>
            <Input
              type="text"
              size="large"
              placeholder={this.props.placeholderText}
              value={this.state.search}
              onChange={e => {
                this.setState({ search: e.target.value });
                if (this.props.handleTextChanged) {
                  this.props.handleTextChanged();
                }
              }}
            />
            <Button type="primary" onClick={e => this.handleFormSubmit(e)}>
              <Icon type="search" />
              {this.props.buttonText}
            </Button>
          </FormItem>
        </Form>
      </Card>
    );
  }
}

SearchForm.defaultProps = {
  placeholderText: 'ใส่คำค้นหา...',
  buttonText: 'ค้นหา',
};

function mapStateToProps(state) {
  return { ...state };
}

export default connect(mapStateToProps)(SearchForm);
