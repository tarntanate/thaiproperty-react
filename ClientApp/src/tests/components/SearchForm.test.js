import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";

import SearchForm from '../../components/Shared/SearchForm';

// Mocking objects
const mockStore = configureMockStore();
const store = mockStore({});

describe('SearchForm Components', () => {
  it('Should renders correctly', () => {
    const tree = renderer
      .create(<Provider store = {store}>
                <SearchForm />
              </Provider>)
      .toJSON();
    console.log(tree);
    expect(tree).toMatchSnapshot();
  });
});