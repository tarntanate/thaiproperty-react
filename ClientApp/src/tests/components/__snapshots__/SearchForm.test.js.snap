// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm Components Should renders correctly 1`] = `
<div
  className="ant-card"
>
  <div
    className="ant-card-body"
    style={undefined}
  >
    <form
      className="ant-form ant-form-inline"
      onSubmit={[Function]}
    >
      <div
        className="ant-row ant-form-item"
        style={undefined}
      >
        <div
          className="ant-col-24 ant-form-item-control-wrapper"
        >
          <div
            className="ant-form-item-control"
          >
            <span
              className="ant-form-item-children"
            >
              <input
                className="ant-input ant-input-lg"
                disabled={false}
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="ใส่คำค้นหา..."
                size="large"
                type="text"
                value=""
              />
              <button
                className="ant-btn ant-btn-primary"
                onClick={[Function]}
                type="button"
              >
                <i
                  className="anticon anticon-search"
                />
                <span>
                  ค้นหา
                </span>
              </button>
            </span>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
